### PR TITLE
Issue 209: Support rename log

### PR DESCRIPTION
**Pull request by sijie**
Descriptions of the changes in this PR:

- add rename operation in `Namespace`
- add rename operation in `LogStreamMetadataStore`
- implement the rename operation use zookeeper `multi` operation

### CODE REVIEW

File: distributedlog-core/src/main/java/org/apache/distributedlog/impl/metadata/ZKLogStreamMetadataStore.java [hunk 1]
```diff
@@ -620,4 +631,271 @@ public void processResult(int rc, String path, Object ctx) {
...
+            createOps, deleteOps
+        );
+
+        // 3. version
+        createOps.addLast(Op.create(
+            newRootPath + VERSION_PATH, intToBytes(LAYOUT_VERSION), acls, CreateMode.PERSISTENT));
+        deleteOps.addFirst(Op.delete(
+            oldRootPath + VERSION_PATH, -1));
+
+        // 4. lock path (NOTE: if the stream is locked by a writer, then the delete will fail as you can not
```
Comment (jiazhai) [inline distributedlog-core/src/main/java/org/apache/distributedlog/impl/metadata/ZKLogStreamMetadataStore.java]: The rename operation will fail if the stream is still locked, right?  
If it is not complex, how about also add a test cases for this situation?

**Comment by sijie** [comment]
@jiazhai I added a test case for renaming a locked log.

File: distributedlog-core/src/main/java/org/apache/distributedlog/impl/metadata/ZKLogStreamMetadataStore.java (hunk 1 above)
Comment (sijie) [reply distributedlog-core/src/main/java/org/apache/distributedlog/impl/metadata/ZKLogStreamMetadataStore.java]: done

**Comment by jiazhai** [comment]
Thanks for adding the test case. LGTM.

**Pull request closed without merging**